### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by kidonng**
A long-overdue thanksgiving gift 🦃️

## Screenshot

- Fix `file-finder-buffer` left margin

Before:

屏幕录制2021-11-29 20 25 06-0001

After:

屏幕录制2021-11-29 20 25 35-0002

### CODE REVIEW

File: source/features/clean-rich-text-editor.tsx [hunk 1]
```diff
@@ -1,30 +1,6 @@
...
-import onetime from 'onetime';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
-function hideButtons(): void {
-	document.body.classList.add('rgh-clean-rich-text-editor');
-}
-
-function hideTextareaTooltip(): void {
```
Comment (kidonng) [inline source/features/clean-rich-text-editor.tsx:12]: This is [one of my earliest contributions](https://github.com/refined-github/refined-github/pull/2375) to RGH 2 years ago. Time flies 🚀

File: package.json [hunk 2]
```diff
@@ -126,7 +126,7 @@
 			"firefoxProfile": "./test/web-ext-profile",
 			"chromiumProfile": "./test/web-ext-profile",
 			"startUrl": [
-				"github.com"
+				"https://github.com"
```
Comment (fregante) [inline package.json:129]: There are more features visible here

```suggestion
				"https://github.com/refined-github/refined-github"
```

File: source/features/clean-rich-text-editor.tsx (hunk 1 above)
Comment (fregante) [reply source/features/clean-rich-text-editor.tsx:12]: No more `title` anywhere? 🎉

**Review by fregante — APPROVED** [review]

File: source/features/clean-rich-text-editor.tsx (hunk 1 above)
Comment (kidonng) [reply source/features/clean-rich-text-editor.tsx:12]: It really just selects comment boxes which doesn't have `title` anymore.

File: package.json (hunk 2 above)
Comment (cheap-glitch) [reply package.json:129]: I'd rather start on GitHub's homepage, it's easier to jump to recent issues/PRs that way.

File: source/features/index.tsx [hunk 3]
```diff
@@ -59,11 +59,15 @@ const logError = (url: string, error: unknown): void => {
 	// Don't change this to `throw Error` because Firefox doesn't show extensions' errors in the console
 	console.group('❌', id, version, pageDetect.isEnterprise() ? 'GHE →' : '→', error);
 
+	const searchIssueUrl = new URL('https://github.com/refined-github/refined-github/issues');
+	searchIssueUrl.searchParams.set('q', `is:issue "${message}"`);
```
Comment (kidonng) [inline source/features/index.tsx:63]: Not sure if quoting them is better, at least that's how it looks to me.

File: source/features/index.tsx [hunk 4]
```diff
@@ -59,11 +59,15 @@ const logError = (url: string, error: unknown): void => {
...
 	console.group('Search issue');
-	console.log(`https://github.com/refined-github/refined-github/issues?q=is%3Aissue+${encodeURIComponent(message)}`);
+	console.log(searchIssueUrl.href);
 	console.groupEnd();
 
-	const newIssueUrl = new URL('https://github.com/refined-github/refined-github/issues/new?labels=bug&template=1_bug_report.md');
+	const newIssueUrl = new URL('https://github.com/refined-github/refined-github/issues/new');
+	newIssueUrl.searchParams.set('labels', 'bug');
+	newIssueUrl.searchParams.set('template', '1_bug_report.yml');
```
Comment (kidonng) [inline source/features/index.tsx:70]: Yup, a bug of bug report (`.md` -> `.yml`) 😄
You cannot click this line (here, with `linkify-url` or your editor) to create a bug report anymore but who would do that anyway.

File: package.json (hunk 2 above)
Comment (fregante) [reply package.json:129]: > it's easier to jump to recent issues/PRs that way

I don't see how that's possible

File: source/features/index.tsx (hunk 4 above)
Comment (fregante) [reply source/features/index.tsx:70]: Does it work with the new issue form though? Does it auto-fill?

File: source/features/deprioritize-marketplace-link.tsx [hunk 5]
```diff
@@ -1,30 +1,28 @@
...
-			<div className="dropdown-divider"/>,
-			<a role="menuitem" className="dropdown-item" href="/marketplace">Marketplace</a>,
-		);
+	const marketplaceLink = select('.Header-link[href="/marketplace"]');
+	// On GHE it can be disabled #3725
+	if (!marketplaceLink) {
+		return;
 	}
+
+	// The link seems to have an additional wrapper that other links don't have https://i.imgur.com/KV9rtSq.png
```
Comment (fregante) [inline source/features/deprioritize-marketplace-link.tsx:14]: We can drop this, this feature doesn't affect any other link, so their DOM is irrelevant

File: source/features/index.tsx [hunk 6]
```diff
@@ -59,11 +59,15 @@ const logError = (url: string, error: unknown): void => {
...
+	const searchIssueUrl = new URL('https://github.com/refined-github/refined-github/issues');
+	searchIssueUrl.searchParams.set('q', `is:issue is:open sort:updated-desc "${message}"`);
 	console.group('Search issue');
-	console.log(`https://github.com/refined-github/refined-github/issues?q=is%3Aissue+${encodeURIComponent(message)}`);
+	console.log(searchIssueUrl.href);
 	console.groupEnd();
 
-	const newIssueUrl = new URL('https://github.com/refined-github/refined-github/issues/new?labels=bug&template=1_bug_report.md');
+	const newIssueUrl = new URL('https://github.com/refined-github/refined-github/issues/new');
+	newIssueUrl.searchParams.set('labels', 'bug');
```
Comment (fregante) [inline source/features/index.tsx:69]: This should be handled by the form already

File: source/features/linkify-user-location.tsx [hunk 7]
```diff
@@ -12,10 +12,17 @@ function addLocation(baseElement: HTMLElement): void {
 		}
 
 		const locationName = location.textContent!.trim();
-		const googleMapsLink = `https://www.google.com/maps/search/?api=1&query=${encodeURIComponent(locationName)}`;
+		const googleMapsLink = `https://www.google.com/maps/search/?api=1&query=${locationName}`;
```
Comment (fregante) [inline source/features/linkify-user-location.tsx:15]: Why not? What if their location is `nah&somethingelse=do`?

**Review by fregante — APPROVED** [review]
LGTM so far other than the open reviews. I'm reviewing as we go 😅

File: source/features/index.tsx (hunk 6 above)
Comment (kidonng) [reply source/features/index.tsx:69]: Nope, try it yourself: https://github.com/refined-github/refined-github/issues/new?template=1_bug_report.yml

image

File: package.json (hunk 2 above)
Comment (cheap-glitch) [reply package.json:129]: Sorry I meant the dashboard :man_facepalming:

Comment (kidonng) [reply package.json:129]: Isn't there a "Recent activity" section on the left side of dashboard?

File: source/features/deprioritize-marketplace-link.tsx (hunk 5 above)
Comment (kidonng) [reply source/features/deprioritize-marketplace-link.tsx:14]: Dp you mean just remove the link directly?

Comment (kidonng) [reply source/features/deprioritize-marketplace-link.tsx:14]: I'm moving `deprioritize-marketplace-link` to a separate PR

File: source/features/index.tsx (hunk 4 above)
Comment (kidonng) [reply source/features/index.tsx:70]: Not yet but I'm fixing it

File: source/features/linkify-user-location.tsx (hunk 7 above)
Comment (kidonng) [reply source/features/linkify-user-location.tsx:15]: Good call, reverted

File: source/features/index.tsx (hunk 6 above)
Comment (fregante) [reply source/features/index.tsx:69]: Try it yourself 😉 

Non-maintainers cannot specify labels, only the template can. What those parameters do is change the immediate view, for example the title isn't automatically set even if the template includes it.

You can see that the labels are missing from the sidebar, but they are added anyway.

Original link: https://github.com/typescript-eslint/typescript-eslint/issues/new?assignees=&labels=enhancement%3A+new+base+rule+extension%2Cpackage%3A+eslint-plugin%2Ctriage&template=eslint-plugin-typescript.yml&title=Rule+proposal%3A+

Link I used for this video: https://github.com/typescript-eslint/typescript-eslint/issues/new?template=eslint-plugin-typescript.yml

https://user-images.githubusercontent.com/1402241/144068558-84e17e02-ef73-4c9a-8e95-f38ca619f784.mov

File: source/features/index.tsx [hunk 8]
```diff
@@ -69,15 +69,15 @@ const logError = (url: string, error: unknown): void => {
...
 		<!-- Please also include a screenshot if the issue is visible -->
 
-		URL: ${location.href}
-
 		\`\`\`
 		${error instanceof Error ? error.stack! : error as string}
 		\`\`\`
 	`));
+	newIssueUrl.searchParams.set('example_url', location.href);
+	newIssueUrl.searchParams.set('browser_s_used', navigator.userAgent);
```
Comment (fregante) [inline source/features/index.tsx:80]: NIIICE.

File: source/features/index.tsx (hunk 6 above)
Comment (fregante) [reply source/features/index.tsx:69]: Damn GitHub 😂 https://github.com/refined-github/refined-github/issues/5135

Let's keep the `labels` parameter

Comment (kidonng) [reply source/features/index.tsx:69]: _[Confused Pikachu]_

File: source/features/index.tsx (hunk 8 above)
Comment (kidonng) [reply source/features/index.tsx:80]: > Also `example_url` can be moved above `description`

It is following the order of the template

File: source/features/index.tsx [hunk 9]
```diff
@@ -77,7 +77,10 @@ const logError = (url: string, error: unknown): void => {
 		\`\`\`
 	`));
 	newIssueUrl.searchParams.set('example_url', location.href);
-	newIssueUrl.searchParams.set('browser_s_used', navigator.userAgent);
+	// @ts-expect-error navigator.userAgentData is not typed
+	const {platform} = navigator.userAgentData || navigator;
+	const browser = /(Chrome|Firefox|Safari)\/\d+/.exec(navigator.userAgent)?.[0];
+	newIssueUrl.searchParams.set('browser_s_used', [platform, browser].join(' '));
```
Comment (kidonng) [inline source/features/index.tsx:83]: Not jumping into the rabbit hole but this should be enough (platform/OS name + browser name + major browser version).

"Chrome" here matches any Chromium-based browser, since we only care about _its_ version.

As an example, it produces `macOS Chrome/96`, `MacIntel Firefox/94` and `MacIntel Safari/605` on my machine.

File: package.json (hunk 2 above)
Comment (fregante) [reply package.json:129]: > Isn't there a "Recent activity" section on the left side of dashboard?

I see 4 links to issues unrelated to RGH, I think RGH’s repo is a better starting point to access RGH’s issues. Maybe even open the issues directly

```suggestion
				"https://github.com/refined-github/refined-github/issues"
```

File: source/features/index.tsx (hunk 3 above)
Comment (fregante) [reply source/features/index.tsx]: Maybe not, issues may still be useful even if they're not identical

File: source/features/index.tsx (hunk 6 above)
Comment (fregante) [reply source/features/index.tsx:69]: Good as-is

File: source/features/index.tsx [hunk 10]
```diff
@@ -59,21 +59,33 @@ const logError = (url: string, error: unknown): void => {
...
 	`));
+	newIssueUrl.searchParams.set('example_url', location.href);
+
+	const browser = /(Chrome|Firefox|Safari)\/\d+/.exec(navigator.userAgent);
+	// Unknown browsers will need to be specified manually
+	if (browser) {
+		// @ts-expect-error navigator.userAgentData is not typed
+		const {platform} = navigator.userAgentData || navigator;
+		newIssueUrl.searchParams.set('browser_s_used', `${platform as string} ${browser[0]}`);
+	}
```
Comment (fregante) [inline source/features/index.tsx:87]: As [mentioned before](https://github.com/refined-github/refined-github/pull/5113#discussion_r759351014), but we should not do this. Let the user enter the browser, which may be reported as "chrome" when in reality it's Vivaldi. We don't need to parse this information because it's so fickle 

```suggestion
```

File: source/features/split-issue-pr-search-results.tsx [hunk 11]
```diff
@@ -28,7 +28,11 @@ function updateLinkElement(link: HTMLAnchorElement, type: GitHubConversationType
 function init(): void {
 	cleanLinks();
 
-	const issueLink = select('a.menu-item[href*="&type=issues"]')!;
+	const issueLink = select('a.menu-item[href*="&type=issues"]');
+	if (!issueLink) {
+		return;
+	}
```
Comment (fregante) [inline source/features/split-issue-pr-search-results.tsx:34]: When does this happen?

File: source/features/index.tsx [hunk 12]
```diff
@@ -59,21 +59,33 @@ const logError = (url: string, error: unknown): void => {
...
-
 		\`\`\`
 		${error instanceof Error ? error.stack! : error as string}
 		\`\`\`
 	`));
+	newIssueUrl.searchParams.set('example_url', location.href);
+
+	const browser = /(Chrome|Firefox|Safari)\/\d+/.exec(navigator.userAgent);
+	// Unknown browsers will need to be specified manually
+	if (browser) {
```
Comment (yakov116) [inline source/features/index.tsx:83]: I think this should be moved to another PR and have it close https://github.com/refined-github/refined-github/issues/4292

WDYT?

Comment (kidonng) [reply source/features/index.tsx:83]: Alright, I will just link the discussion here.

File: source/features/split-issue-pr-search-results.tsx (hunk 11 above)
Comment (kidonng) [reply source/features/split-issue-pr-search-results.tsx:34]: I mentioned this in the commit message: https://github.com/refined-github/refined-github/search

**Comment by fregante** [comment]
I would not add actual changes in lint PRs because they get lost and there's no argument/screenshot for the change. Lint PRs are for refactorings and code cleanups, not bugfixes nor new features

**Comment by yakov116** [comment]
> because they get lost

I learned this the hard way :+1:

File: source/features/split-issue-pr-search-results.tsx (hunk 11 above)
Comment (fregante) [reply source/features/split-issue-pr-search-results.tsx:34]: We should use a URL-based exclusion instead because this silences any future errors we might have. Example:

```js
eclude: [
	() => location.search === ''
]
```

No need for a full, custom detection in github-url-detection

**Review by fregante — APPROVED** [review]
🚢

File: source/features/index.tsx (hunk 3 above)
Comment (kidonng) [reply source/features/index.tsx]: > Maybe not, issues may still be useful even if they're not identical

Somehow I missed see this message 😅

**Comment by fregante** [comment]
Let's not keep lint PRs around for too long. It's good as is for now. You can review the "restore" commits I made and reviews I left and open separate PRs for those.
